README.md

```diff
@@ -1,26 +1,63 @@
-# Terraform Module Template
-
-**Next steps**
-1. Update the top section of this file to tell people about this module.
-2. Update `versions.tf` to include the required providers for the module.
-3. Add resources and variables to solve the problem.
-4. Add outputs for relevant details the consumer may want
-5. Add example uses to the bottom of this file
-6. Update the generated portion of this file using `terraform-docs .`
-
+# gcp-terraform-drata-setup
+
+GCP terraform module to create the Drata Read Only service account.
+
+## Usage
+
+The example below uses `ref=main` (which is appended in the URL),  but it is recommended to use a specific tag version (i.e. `ref=1.0.0`) to avoid breaking changes. Go to the release page for a list of published versions. [releases page](https://github.com/drata/gcp-terraform-drata-setup/releases) for a list of published versions.
+
+Replace `YOUR_ORGANIZATION_DOMAIN` with the organization domain. i.e. `your_org.com`.
+```
+module "service_account_creation" {
+  source = "git::https://github.com/drata/gcp-terraform-drata-setup.git?ref=main"
+  gcp_org_domain = "YOUR_ORGANIZATION_DOMAIN"
+  # gcp_project_id = "YOUR_PROJECT_ID" # if it's unset, the project by default is used
+  # drata_role_name = "YOUR_ROLE_NAME" # if it's unset, the default name is DrataReadOnly
+}
+
+output "drata_service_account_key" {
+  value = module.service_account_creation.drata_service_account_key
+  description = "Service Account Key"
+  sensitive = true
+}
+```
+
+After you apply this terraform, run the following command to retrieve the key file `drata-gcp-private-key.json`
+```
+terraform output -raw drata_service_account_key > drata-gcp-private-key.json
+```
+
+## Setup
+
+The following steps demonstrate how to connect GCP in Drata when using this terraform module.
+
+1. Add the code above to your terraform project.
+2. Make sure the service account to authenticate this script has the roles `Organization Administrator`, `Service Account Admin`, `Service Account Key Admin` and ` Service Usage Admin`.
+3. Replace `main` in `ref=main` with the latest version from the [releases page](https://github.com/drata/gcp-terraform-drata-setup/releases).
+4. Replace `YOUR_ORGANIZATION_DOMAIN` with the GCP organization domain.
+5. You could replace `YOUR_PROJECT_ID` with the project ID optionally otherwise the default project is utilized.
+6. `drata_role_name` could also be replaced otherwise the default name is `DrataReadOnly`.
+7. Back in your terminal, run `terraform init` to download/update the module.
+8. Run `terraform apply` and **IMPORTANT** review the plan output before typing `yes`.
+9. If success, run the command to generate the json key file 
+     - `terraform output -raw drata_service_account_key > drata-gcp-private-key.json` .
+11. Verify the file is created in your end.
+12. Go to the GCP connection drawer for user access review and select Upload File to upload the `drata-gcp-private-key.json` file.
```

Comment:
Why "for user access review" ?

Reply:
yeah we can review 'for user access review' from this sentence

Reply:
> Why "for user access review" ?

that was a little confusion, that should say `Go to the GCP connection drawer for infrastructure...` otherwise just `Go to the GCP connection drawer...`

Reply:
[done](https://github.com/drata/gcp-terraform-drata-setup/pull/2/commits/e3978f71da58392dc59871333d5ccaa492701e72) 🚀